lib/msal-common/src/client/BaseClient.ts

```diff
@@ -136,4 +138,22 @@ export abstract class BaseClient {
         }
         this.authority = updatedAuthority;
     }
+
+    /**
+     * Creates query string for the /token request
+     * @param request
+     */
+    createTokenQueryParameters(request:
```

Comment:
Can't this accept a param `request` of type `BaseAuthRequest` ?

This design is a code smell - a base class should not know about derived classes. Otherwise, the design becomes `brittle`, i.e. when a new subclass is defined, you must also update the base class. 

See the O and L principles from SOLID - https://www.baeldung.com/solid-principles

Reply:
good point, thanks